### ISSUE TITLE
Wrong highlighting on here-string (<<<)

### ISSUE THREAD

**Issue by de-jcup** [de-jcup/eclipse-bash-editor#90]
``` bash
text="a b c"
read a b c<<<"$text
echo $a $b $c  # actual code, no here-document!
```
is highlighted wrong:

![image](https://user-images.githubusercontent.com/11644753/33941896-7902f13c-e014-11e7-834f-ed181f2d001d.png)

see 
- http://tldp.org/LDP/abs/html/x17837.html
- https://unix.stackexchange.com/questions/76402/command-line-instead-of 
- https://www.gnu.org/software/bash/manual/bashref.html#Here-Strings
for information

Origin information came from https://marketplace.eclipse.org/comment/5055#comment-5055

**Comment by de-jcup** [comment]
When lookin inside http://tldp.org/LDP/abs/html/x17837.html and trying to execute the example script from above it seems that the script has got itself a failure because of wrong line:
`read a b c<<<"$text`
which should be replaced by:
`read a b c<<<"$text"`

Never the less I will fix up the her-doc handling here: when << is followed by a < it will not be recognized as here-doc any longer.

**Comment by de-jcup** [comment]
Never the less the parser did have problems with determinating <<< as herestring instead of heredoc. This was fixed in last commit above.

**Comment by de-jcup** [comment]
Hello @TheMagican,
could you please look at the problem described above? In my opinion it is correct to handle the multiline string as multiline.  I changed the test script to

``` bash
text="a b c"
read a b c<<<"$text
echo $a $b $c  # actual code, no here-document!"
echo $a $b $c # this is really actual code
echo "Next echo statement..."

```
and got following:

![image](https://user-images.githubusercontent.com/11644753/33986261-5dc559a6-e0bd-11e7-8831-f270ae2b1fc4.png)

I think this correct.

**Comment by de-jcup** [comment]
@TheMagican : Thank you!

**Comment by zeroincombenze** [comment]
Sorry, I posted a wrong example
This is th right code to replicate error
```
get_full_xtlcmd() {
    local xtlcmd=$(get_act_xtlcmd $2 "$1" "$3" "$4" "$5")
    local pkg lop lreqver p rop rreqver branch x pkgURL subpkg
    IFS="~" read pkg lop lreqver rop rreqver branch x<<<"$5"
    p=" $2"
    if [ "$2" == "EVERYTHINGS" ]; then
        p=""
    fi
    pkgURL=$(get_full_URL "$1" "$2" "$3" "$4" "$5")
    [ -n "$pkgURL" ] && p=" $pkgURL"
```

**Comment by zeroincombenze** [comment]
FYI,
in most pieces of bash code recognition works well

**Comment by de-jcup** [comment]
Hello @zeroincombenze ,

as you mentioned syntax highlighting for your second script is odd (Bash editor V1.4.0)
![image](https://user-images.githubusercontent.com/11644753/34050072-d5e03a6c-e1b9-11e7-9163-34233c97534d.png)

The next version will handle it correct:
![image](https://user-images.githubusercontent.com/11644753/34050226-7647f224-e1ba-11e7-8981-d3b5e0d8b198.png)

You can also define a custom color for here-strings now: 
![image](https://user-images.githubusercontent.com/11644753/34050303-ae95c66a-e1ba-11e7-815b-d81893368ede.png)

**Comment by zeroincombenze** [comment]
Hi,
thanks for your quick and clear response.
I have a question: bash editor does have some limits?
Look at this screen-shot:
![immagine](https://user-images.githubusercontent.com/6972555/34069829-6082887e-e259-11e7-9cc3-8c91b28dd336.png)
Bash edito show an error I cannot understand

Here another strange error:
![immagine](https://user-images.githubusercontent.com/6972555/34069854-b2eed176-e259-11e7-8015-4130c51103b6.png)

and here the strangest error:
![immagine](https://user-images.githubusercontent.com/6972555/34069870-e5d5a934-e259-11e7-998b-afadc827ecb6.png)

Notice, if extract above code errors die!
Because my bash script is very complex and has many lines, I guess bash editor goes in overflow in some counter during parsing.

However my codind is simplified with your bash editor. 
Thanks! You have done a great job!

**Comment by de-jcup** [comment]
@zeroincombenze 
First of all: you are wllcome :-)

I created a new issue #91 for your script validation problems.
I will copy your last entry to #91 - it does not belong to #90.

The next version has no longer here-string problems so I close #90 now.

**Comment by de-jcup** [comment]
@zeroincombenze I released Version 1.4.1 (because of another bug instead of Version 1.5.0 ). This version contains the bugfix for herestring

**Comment by zeroincombenze** [comment]
Hello,
I confirm 1.4.1 solved trouble. And solved 2 of problems of #91 
You are Great!
Thsnks
